### ISSUE TITLE
Potential Data Leakage: Validation scenes seems included in SFT Cold Start Dataset in `scenecold_dataset.py`

### ISSUE THREAD

**Issue by PrikettoLiddel** [AIGeeksGroup/3D-R1#44]
I would like to appreciate for your valuable and inspring work on 3D-R1. And I appreciate the effort put into open-sourcing the codebase for the community.

However, while studying the code implementation to better understand the data pipeline, I noticed a discrepancy in the [dataset/scenecold_dataset.py](https://github.com/AIGeeksGroup/3D-R1/blob/main/dataset/scenecold_dataset.py) file that seems to undermine the validity of the training/validation split:
https://github.com/AIGeeksGroup/3D-R1/blob/767a5e2e10bc57d887fa5f92a8fc0bcd5df4050f/dataset/scenecold_dataset.py#L44-L52

In the `Dataset` class initialization, although the parent class `ScanNetBaseDataset` correctly loads the scan names based on the requested split_set (e.g., 'train'), the child class unconditionally overwrites this list with all scenes present in the annotation file `Scene-30K.jsonl`. From the dataset file you uploaded at [HuggingFace](https://huggingface.co/datasets/AIGeeksGroup/Scene-30K), it seems that validation scenes are also included in it (e.g, scene with id `scene0011_00`).

Would this lead to data leakage/unfairness of training on validation scenes in ScanNet? Especially when downstream tasks, like ScanRefer, Nr3D, SQA3D and ScanQA are tested on validation scenes. I hope this helps in revisiting the experiments to ensure the integrity of the evaluation results.

**Comment by Believeht029** [comment]
Thanks a lot for the thoughtful read and for sharing the concern—it’s an important point to double‑check.
*For clarity:* `Scene-30K` is only used as a cold-start SFT corpus to warm up reasoning. That stage never reports metrics, and we intentionally treat the whole JSONL (train + val scene IDs) as a synthetic pretraining set. When we move to official benchmarks (ScanRefer, Nr3D, SQA3D, ScanQA, etc.), we rebuild the datasets via dataset/unified_*.py, which load annotations and scan lists strictly from the split files (ScanRefer_filtered_train.txt, nr3d_train.txt, scannetv2_train.txt, etc.). Those loaders overwrite self.scan_names with the canonical split lists inside ScanNetBaseDataset, so only the right scenes feed the actual training/eval runs.
In short, even if `Scene-30K.jsonl` contains IDs that also appear in the downstream validation sets, it functions like any external pretraining corpus (similar to using ImageNet train+val before fine-tuning on COCO). The downstream fine-tuning and all reported validation results rely solely on the standard ScanNet splits, so there isn’t leakage into the evaluation metrics.

**Comment by PrikettoLiddel** [comment]
Thanks for the quick and detailed clarification! I understand the rationale of treating `Scene-30K` as synthetic pretraining corpus to "warm up" the reasoning capabilities.

However, I believe there is a subtle but critical distinction here compared to general pretraining. In the case of ImageNet vs. COCO, the pretraining model sees images but not the specific labels or ground-truth annotations of the downstream test set.

In the case of Scene-30K, it appears to contain the exact (or very similar) Questions and Answers from the validation sets of the downstream benchmarks. E.g., I noticed that `Scene-30K.jsonl` explicitly includes data derived from validation scenes. For instance, Line 25564 corresponds to scene0011_00 (which is a validation scene in ScanNet/ScanQA):

If I didn't understand wrongly, the Scene-30K is built by including data from multiple task data. E.g., it includes ScanQA data's questions (and answers). 

```
// From Scene-30K.jsonl
{
  "scene_id": "scene0011_00", 
  "question": "What color is the chair in the kitchen?", 
  "cot": "<think>...</think><answer>The color of the chair in the kitchen is brown or dark brown.</answer>"
}
```
which is clearly same (or very similar) as the first question in ScanQA's validation set:
```
// From ScanQA
{
  "scene_id": "scene0011_00",
  "question": "What color is the chair in the kitchen?", 
  "answers": ["dark brown", "brown"],
  ...
}
```

Even if metrics are not reported during this SFT stage, the model is effectively being optimized to memorize the correct answers for the validation questions (and since ScanQA questions are built from ScanRefer referring expressions, which are also used to build Scan2Cap dataset, there is still a serious concern for grounding and captioning tasks)

Regardless of similarity of used instructions, in the context of 3D MLLMs (such as LEO, LLaVA-3D, Chat-Scene, etc.), the community standard seems to be typically to strictly isolate validation scenes from all training stages to ensure that the evaluation measures generalization rather than memorization. It might be unfair if the model have witnessed validation scenes in ScanNet in training stage, making fair comparison with other methods difficult.

It might be worth considering filtering out these validation scenes from Scene-30K to align with the standard strict-split protocols.

Thanks again for the discussion!

**Comment by Believeht029** [comment]
To be absolutely clear: we never use the ScanNet/ScanQA validation split for SFT optimization. The only moment those scenes appear is when we swap the loader to `split_set="val"` to report validation metrics—no gradients flow, no weights are updated there. The SFT stage itself always instantiates `dataset.scenecold_dataset.Dataset(..., split_set="train")`, and that constructor inherits the train-only scan list from `ScanNetBaseDataset`, so the optimizer never even sees the validation files.

**Comment by PrikettoLiddel** [comment]
Thanks for the clarification! I completely understand the intended design to rely on the ScanNetBaseDataset logic to handle the splitting, ensuring strict isolation. That makes perfect sense architecturally. However, I suspect there might be a subtle implementation bug in [scenecold_dataset.py](https://github.com/AIGeeksGroup/3D-R1/blob/main/dataset/scenecold_dataset.py) that went unnoticed, which inadvertently bypasses this intended logic:

While `super().__init__` indeed sets up the correct, filtered `self.scan_names`, it appears that lines:
https://github.com/AIGeeksGroup/3D-R1/blob/767a5e2e10bc57d887fa5f92a8fc0bcd5df4050f/dataset/scenecold_dataset.py#L44-L52 in the child class accidentally overwrites this variable, breaking the inheritance contract, and accidentally using all samples from Scene-30K.

Could you please double-check this specific execution flow? Thanks again for the useful discussion!

**Comment by Believeht029** [comment]
Thanks for flagging it. Internally we maintained two JSONL shards (`Scene-30K_train.jsonl` and `Scene-30K_val.jsonl`) and the training script only ever mounted the train shard, so the reported numbers were produced with a strict split. When we cleaned everything up for release we merged the shards into the single `Scene-30K.jsonl` to simplify hosting, but we forgot to port the filtering logic that kept validation scenes out of the train loader—so the public file plus the current scenecold_dataset.py do indeed let every scene_id through.
We’re fixing that right now in two steps: (1) reintroduce the filtering so that self.annotations is intersected with the parent’s scan list (ensuring train stays train, val stays val) and (2) upload the split JSONLs alongside the merged version so anyone can reproduce our exact setup. The training protocol in the paper already used the split version, so the published results aren’t affected, but we agree the released code should enforce it as well. 
Thanks again for catching the discrepancy—we’ll push the patch and the extra files shortly.

**Comment by PrikettoLiddel** [comment]
Sorry for the late reply, thank you for your clarifications!
